File: data/code_pairs/fortran/1083.f90
program loop_hoist6
  call clawloop
end program loop_hoist6

subroutine clawloop
  implicit none
  integer :: i, j, k
  integer, parameter :: iend = 2, jend = 4, kend = 2

  do i = 0, iend
    if (i == 0) then
      print *, 'First iteration of i', i, '/', j, '/', k
    end if
    do j = 0, jend
      do k = 0, kend
        print *, 'First loop body:', i, '/', j, '/', k
      end do
    end do

    do j = 2, jend
      do k = 0, kend
        print *, 'Second loop body:', i, '/', j, '/', k
      end do
    end do

    do j = 0, jend
      print *, 'ALONE J LOOP'
    end do
  end do
end subroutine clawloop

File: data/code_pairs/cpp/1083.cpp
#include <iostream>

void clawloop();

int main() {
    clawloop();
    return 0;
}

void clawloop() {
    int i, j, k;
    const int iend = 2;
    const int jend = 4;
    const int kend = 2;

    for (i = 0; i <= iend; ++i) {
        if (i == 0) {
            std::cout << "First iteration of i " << i << '/' << j << '/' << k << std::endl;
        }
        for (j = 0; j <= jend; ++j) {
            for (k = 0; k <= kend; ++k) {
                std::cout << "First loop body: " << i << '/' << j << '/' << k << std::endl;
            }
        }

        for (j = 2; j <= jend; ++j) {
            for (k = 0; k <= kend; ++k) {
                std::cout << "Second loop body: " << i << '/' << j << '/' << k << std::endl;
            }
        }

        for (j = 0; j <= jend; ++j) {
            std::cout << "ALONE J LOOP" << std::endl;
        }
    }
}